src/librustc/infer/error_reporting/nice_region_error/util.rs

```diff
@@ -12,7 +12,7 @@ use syntax_pos::Span;
 #[derive(Debug)]
 pub(super) struct AnonymousArgInfo<'tcx> {
     // the argument corresponding to the anonymous region
-    pub arg: &'tcx hir::Arg,
+    pub arg: &'tcx hir::Param,
```

Comment:
I think this should be `param` and the whole struct should be `AnonymousParamInfo`.